tripy/tripy/frontend/utils.py

```diff
@@ -68,86 +185,13 @@ def convert_inputs_to_tensors(
     sync_arg_types = utils.default(sync_arg_types, [])
     exclude = utils.default(exclude, [])
     unpack_argument = utils.default(unpack_argument, [])
-    shape_argument = utils.default(shape_argument, [])
 
     def impl(func):
```

Comment:
Could we assert that `func` is a magic method? We should disallow this decorator for most APIs.

Reply:
It would be easy to check the name, but there are exceptions in the codebase:
* `quantize` and `dequantize`
* The conversion for slicing is done in `slice_helper`
* `minimum` and `maximum` are binary elementwise in TRT, but they're not Python operators

We could encode the exceptions explicitly (which would thus discourage the use of the decorator in cases other than operator overloads) or manually do conversions in these cases. Doing manual conversions might make sense other than for `slice_helper`, though it has the disadvantage of not filling in as much stack information.

Reply:
`quantize`, `dequantize`, `minimum`, and `maximum` shouldn't be doing automatic conversions IMO. `slice_helper` is really just a hack since our decorator doesn't handle slices correctly. We could add an exception for it now and eventually fix it in the decorator so we don't need the helper.

Reply:
Put in a simple check, though I had to change the tests for the decorator (we can add them to the exception list if we would prefer normal-looking names there). For quantize and dequantize, I kept in a (manual) conversion for the scale argument because many call sites passed in single numbers and it makes sense to accept them.

Reply:
@yizhuoz004 do you know why Q/DQ ops aren't implemented using frontend ops? The division part at least could be done in the frontend which would also simplify the trace op and not require us to explicitly convert `scale` to a tensor.

Reply:
There are 2 reasons:
- We want to see Q/DQ ops in Trace IR, if implemented using frontend ops, we lose them.
- More stable and concise for pattern matching, with frontend ops, the number of stablehlo ops will explode because of the binary elementwise shape comparisons.

Reply:
Are there any further concerns remaining with the decorator?

Reply:
Looks good now